### ISSUE TITLE
Inquiry: Access to token embedding or UMAP representation

### ISSUE THREAD

**Issue by 497662892** [gerstung-lab/Delphi#24]
Dear authors,

Thank you for releasing such an excellent and inspiring work!

Would it be possible to share the **token embedding matrix (e.g. model.transformer.wte.weight)** or the **full corresponding UMAP projection csv** used for visualization?

This would greatly help us in understanding and extending your findings.

Thank you again for your outstanding contribution and open sharing!

Best

**Comment by yozhikoff** [comment]
Dear @497662892 

Thank you for your interest, and sorry for the delayed response!

The file below contains the 2d UMAP coordinates. Would that be enough for you to use?

[delphi_labels_chapters_colours_icd_umap.csv](https://github.com/user-attachments/files/23343868/delphi_labels_chapters_colours_icd_umap.csv)

**Comment by 497662892** [comment]
Dear @yozhikoff 

Thank you so much! That’s very helpful.
If possible, could I also get access to the token embedding matrix (e.g., model.transformer.wte.weight)? That would be even better! 

Best

**Comment by HelloWorldLTY** [comment]
May I ask if the embeddings of each patients used to plot extended data figure 5 can also be accessed via this method or not? Thanks.

**Comment by yozhikoff** [comment]
@HelloWorldLTY 
Could you please clarify which figure you are referring to? the current extended data figure 5 doesn't mention patient embeddings

**Comment by HelloWorldLTY** [comment]
I think Extended Data Figure 5 allows delphi prediction for disease risk with other metadata? Can I have some references about how to reproduce it? My understanding is you use a linear model with patient input+prs or similar things as input? Thanks.

**Comment by yozhikoff** [comment]
@HelloWorldLTY 
Exactly, it's an extremely simple proof-of-concept approach.
One simply takes the Delphi-predicted disease rate (logits) for the disease of interest, then concatenates this scalar value with the additional features (PRS, blood counts, etc) and then trains a linear model on top of it. 

The exact data used is described in the methods section of the paper "Training of linear models with polygenic risk scores, biomarkers and overall health rating status".

I'm happy to discuss this in more detail, but let's create a separate issue in this case.

**Comment by HelloWorldLTY** [comment]
Thank you, I will do it.
